sympy/solvers/ode/single.py

```diff
@@ -709,3 +709,43 @@ def _ode_factorable_match(eq, func):
             return True
         else:
             return False
+    def _get_general_solution(self, *, simplify: bool = True):
+        func = self.ode_problem.func.func
+        r"""
+        Solves equations having a solvable factor.
+
+        This function is used to solve the equation having factors. Factors may be of type algebraic or ode. It
+        will try to solve each factor independently. Factors will be solved by calling dsolve. We will return the
+        list of solutions.
+
+        Examples
+        ========
+
+        >>> from sympy import Function, dsolve, Eq, pprint, Derivative
+        >>> from sympy.abc import x
+        >>> f = Function('f')
+        >>> eq = (f(x)**2-4)*(f(x).diff(x)+f(x))
+        >>> pprint(dsolve(eq, f(x)))
+                                        -x
+        [f(x) = 2, f(x) = -2, f(x) = C1*e  ]
+
+
+        """
+        eqns = self.check['eqns']
+        sols = []
+        for eq in eqns:
+            try:
+                sol = dsolve(eq, func)
```

Comment:
Now it shows `TypeError: 'property' object is not iterable` for `func` in _deutils.py for this line.

Reply:
What is func here? What should it be instead?

Reply:
here is func is property and it should be` Function`

Reply:
It should be `f(x)` rather than `f`

Reply:
sorry, my bad it was simple fix `dsolve(sol,func(x))` worked.